pkg/controller/vitesskeyspace/vitesskeyspace_controller.go

```diff
@@ -152,60 +152,91 @@ func (r *ReconcileVitessKeyspace) Reconcile(request reconcile.Request) (reconcil
 	})
 	log.Info("Reconciling VitessKeyspace")
 
-	// Fetch the VitessKeyspace instance
-	vtk := &planetscalev2.VitessKeyspace{}
-	err := r.client.Get(ctx, request.NamespacedName, vtk)
+	handler, err := r.NewReconcileHandler(ctx, request)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return resultBuilder.Result()
-		}
-		// Error reading the object - requeue the request.
 		return resultBuilder.Error(err)
 	}
-	planetscalev2.DefaultVitessKeyspace(vtk)
+	if handler == nil {
+		return resultBuilder.Result()
+	}
+	defer handler.close()
 
-	// Reset status, since that's all out of date info that we will recompute now.
-	oldStatus := vtk.Status
-	vtk.Status = planetscalev2.NewVitessKeyspaceStatus()
+	defer func() {
+		err := handler.updateStatus(ctx)
+		if err != nil {
+			finalResult, finalErr = resultBuilder.Error(err)
+		}
+	}()
 
-	// Idle means the keyspace is not deployed in any cells (there are no tablet pools defined).
-	vtk.Status.Idle = k8s.ConditionStatus(len(vtk.Spec.CellNames()) == 0)
+	err = handler.tsInit(ctx)
```

Comment:
That's a fair point. I'll move it inside both `reconcileX()` functions that need it. It should be safe to call more than once because of the != nil check bail in tsInit.

Reply:
Done.